### ISSUE TITLE
Lasagne networks to ConvertJS?

### ISSUE THREAD

**Issue by fkkcloud** [Lasagne/Lasagne#576]
Hi,

I was not able to find proper place to post this questions. Please give me an insight.

I have Lasagne network and can export network as a list of lists using 
lasagne.layers.get_all_param_values(layer, **tags)

However, I am trying to export this as JSON and use it for feedforward in browser using ConvNetJS. What would be the best way/scenario to do this?

**Comment by f0k** [comment]
Exporting this to JSON shouldn't be a problem. For using it in ConvNetJS, you'll need to define a network of the same structure in ConvNetJS, then assign the weights in correct order. The list of params will go from the lowest layers (near the input) to the highest layers (near the output). For every layer, it first lists the weights, then the bias (if any). Carefully check if the weights for ConvNetJS follow the same layout as for Lasagne; if not, you may need to swap some of the axes.

Hint: If you name all your layers (by passing `name='xyz'` to the layer constructor), then all the weights will have clear names, so you can get them with `[p.name for p in lasagne.layers.get_all_params(network)]`. This will help you to get the order correct, or to export them as a dictionary instead of a list.

If you figure out a way that works, we'd welcome a toy example in [Lasagne/Recipes](https://github.com/Lasagne/Recipes)!

**Comment by fkkcloud** [comment]
Good to hear. thanks for rapid respond.
That makes sense.

below is the current network parameters.
	net1 = NeuralNet(
    layers=[('input', layers.InputLayer),
            ('conv2d1', layers.Conv2DLayer),
            ('maxpool1', layers.MaxPool2DLayer),
            ('conv2d2', layers.Conv2DLayer),
            ('maxpool2', layers.MaxPool2DLayer),
            ('dropout1', layers.DropoutLayer),
            ('dense', layers.DenseLayer),
            ('dropout2', layers.DropoutLayer),
            ('output', layers.DenseLayer),
            ],
    # input layer
    input_shape=(None, 1, 48, 48),
    # layer conv2d1
    conv2d1_num_filters=32,
    conv2d1_filter_size=(5, 5),
    conv2d1_nonlinearity=lasagne.nonlinearities.rectify,
    conv2d1_W=lasagne.init.GlorotUniform(),  
    # layer maxpool1
    maxpool1_pool_size=(2, 2),    
    # layer conv2d2
    conv2d2_num_filters=32,
    conv2d2_filter_size=(5, 5),
    conv2d2_nonlinearity=lasagne.nonlinearities.rectify,
    # layer maxpool2
    maxpool2_pool_size=(2, 2),
    # dropout1
    dropout1_p=0.5,    
    # dense
    dense_num_units=256,
    dense_nonlinearity=lasagne.nonlinearities.rectify,    
    # dropout2
    dropout2_p=0.5,    
    # output
    output_nonlinearity=lasagne.nonlinearities.softmax,
    output_num_units=7,
    # optimization method params
    update=nesterov_momentum,
    update_learning_rate=0.005,
    update_momentum=0.9,
    max_epochs=100,
    verbose=1,
    )

Will start transfer them to ConvNetJS after lunch. will let you know the progress.

**Comment by Hrant-Khachatrian** [comment]
@fkkcloud did you manage to do the transfer?

**Comment by fkkcloud** [comment]
I ended up using ConvNetJS for learning too.

If there is any example later, I hope to follow.
